app/components/ItunesCard/index.js

```diff
@@ -0,0 +1,57 @@
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+import { Card, Tooltip, Button } from 'antd';
+import styled from 'styled-components';
+import { T } from '../T/index';
+import { LinkOutlined } from '@ant-design/icons';
+import { truncate } from 'lodash';
+
+const CustomCard = styled(Card)`
+  && {
+    max-width: 350px;
+    margin: 1rem 0;
+  }
+`;
+
+const CustomAudio = styled.audio`
+  margin: 1rem 0;
+`;
+
+const ItunesCard = ({ onClickAction, itune }) => {
+  const audioRef = useRef();
+
+  return (
+    <CustomCard
+      data-testid="itune-card"
+      hoverable
+      title={itune?.trackName}
+      actions={[
+        <Tooltip key="artistViewUrl" title="View Artist">
+          <Button type="link" target="_blank" href={itune?.artistViewUrl} icon={<LinkOutlined />}></Button>
+        </Tooltip>,
+        <Tooltip key="trackViewUrl" title="View Track">
+          <Button type="link" target="_blank" href={itune?.trackViewUrl} icon={<LinkOutlined />}></Button>
+        </Tooltip>,
+        <Tooltip key="collectionViewUrl" title="View Collection">
+          <Button type="link" target="_blank" href={itune?.collectionViewUrl} icon={<LinkOutlined />}></Button>
+        </Tooltip>
+      ]}
+    >
+      <T id="artist_name" values={{ artistName: itune?.artistName }} />
```

Comment:
don't use optional chaining to hide errors. Use it if you truly don't care if the value is not present. 

For example in If conditions it makes sense .

```
if (itune?.artistName) {
   // some business logic
}
```

```
<If condition={itune?.artistName}>
   <T id="artist_name" values={{ artistName: itune?.artistName }} />
</If>
```
->

<DisplayTrackPropertyIfPresent property="artistName" track={track} />

-> 
```
export const DisplayTrackPropertyIfPresent = ({property, intlId, track, ...props ) => <If condition={track[property]}>
   <T id={intlId} values={{ artistName: track[property] }} />
</If>
```